packages/terra-date-picker/src/react-datepicker/calendar.jsx

```diff
@@ -44,57 +44,204 @@ const isDropdownSelect = (element = {}) => {
     forceShowMonthNavigation: PropTypes.bool,
+    /**
+     * A callback function that is executed when date picker is clicked for dropdown.
+     */
     onDropdownFocus: PropTypes.func,
+    /**
+     * A callback function that is executed when a valid date is selected.
+     */
     onSelect: PropTypes.func.isRequired,
+    /**
+     * A callback function that is executed when a Week number is selected.
+     */
     onWeekSelect: PropTypes.func,
+    /**
+     * Prop to determine whether or not to show the time picker.
+     */
     showTimeSelect: PropTypes.bool,
+    /**
+     * Prop to choose format of time selected.
+     */
     timeFormat: PropTypes.string,
+    /**
+     * Interval between 2 consecutive times on time picker in minutes.
+     */
     timeIntervals: PropTypes.number,
+    /**
+     * A callback function to execute when a valid time is selected.
+     */
     onTimeChange: PropTypes.func,
+    /**
+     * Minimum Value of time that can be selected by user.
+     */
     minTime: PropTypes.object,
+    /**
+     * Maximum Value of time that can be selected by user.
+     */
     maxTime: PropTypes.object,
+    /**
+     * Array to store time values disabled for selection.
+     */
     excludeTimes: PropTypes.array,
+    /**
+     * Prop to open calendar on the given date.
+     */
     openToDate: PropTypes.object,
+    /**
+     * Prop to show dates of next month also.
+     */
     peekNextMonth: PropTypes.bool,
+    /**
+     * Prop to show a scrollable dropdown to choose year on the date picker.
+     */
     scrollableYearDropdown: PropTypes.bool,
+    /**
+     * Prop to store previous selection.
+     */
     preSelection: PropTypes.object,
+    /**
+     * Selected Date Value.
+     */
     selected: PropTypes.object,
+    /**
+     * Mark date picker  to select end of range .
+     */
     selectsEnd: PropTypes.bool,
+    /**
+     * Mark date picker  to select start of range .
+     */
     selectsStart: PropTypes.bool,
+    /**
+     * Prop to show a dropdown to select month in date picker .
+     */
     showMonthDropdown: PropTypes.bool,
+    /**
+     * Prop to show week numbers .
+     */
     showWeekNumbers: PropTypes.bool,
+    /**
+     * Prop to show a dropdown to select year in date picker .
+     */
     showYearDropdown: PropTypes.bool,
+    /**
+     * Date selected by user .
+     */
     startDate: PropTypes.object,
+    /**
+     * Name of button to select current date .
+     */
     todayButton: PropTypes.string,
+    /**
+     * Prop to show short names of weekdays .
+     */
     useWeekdaysShort: PropTypes.bool,
+    /**
+     * Prop to open a separate portal version of date picker .
+     */
     withPortal: PropTypes.bool,
+    /**
+     * Difference between utc and local time.
+     */
     utcOffset: PropTypes.number,
+    /**
+     * Label value for weeks on date picker.
+     */
     weekLabel: PropTypes.string,
+    /**
+     * Year Values to show on dropdown +/- the given value.
```

Comment:
I feel this is a better description: https://github.com/cerner/terra-framework/pull/785/files#diff-b61e75265a2b828194784c52122df59cR54